### DISCUSSION TITLE
Changing method of a call webhook action resets body to default

### DISCUSSION THREAD

**Discussion by rowanseymour** [nyaruka/floweditor#1152] [Ideas]
https://app.intercom.com/a/apps/l6e7tr01/inbox/inbox/155936/conversations/20798300029295

I think I remember us discussing this tho I can't remember why we decided to have it like this, and maybe current behavior is right.. but If you have a custom body, and you toggle the method, it resets the body back to the default.

**Comment by ericnewcomer** [comment]
Ya this was definitely by design. It seems telling the method with a complex body would be a rare thing. In any case they can still cancel and worst case use the revision history. Perhaps we could add some warning when a custom body is removed?

**Comment by rowanseymour** [comment]
The problem is that there's no indication that we've just blown away a users' custom payload. You might change from POST to PUT and be completely unaware the payload has been changed. 

I think I lean toward not messing with the payload and give users a way to explicitly revert to the default payload if they want .

**Comment by nicpottier** [comment]
Ya, kinda agree that it should be a more explicit action to blow it away.

**Comment by ericnewcomer** [comment]
The oddity here is switching to POST with some modification and then back to GET they most likely don't expect a payload (we recently added support for GET bodies as a customer request.

**Comment by nicpottier** [comment]
`GET` is definitely the odd duck here for sure. I kind of want to say that switching to `GET` clears it out to default but otherwise we leave it unchanged? (and switching from `GET` sets the default payload)

**Comment by rowanseymour** [comment]
I think the important thing is to preserve user changes. If they haven't touched the default POST body and they change to GET then clear the body, and if they change back to POST then put the default back. But if they've made changes (method == GET and body != "" or method != GET and body == DEFAULT) then we keep that. Sending a superfluous GET body isn't the end of the world.
